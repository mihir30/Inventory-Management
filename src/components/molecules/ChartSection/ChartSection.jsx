/* eslint-disable jsx-a11y/alt-text */
import React from 'react'
import './ChartSection.scss'
import {
    AreaChart, Area, XAxis, YAxis,
    CartesianGrid, Tooltip
} from 'recharts';

const ChartSection = ({ statusResults }) => {


    let finalObj = {}
    statusResults.forEach((statusResult) => {
        const date = statusResult.fromDate;
        if (finalObj[date]) {
            finalObj[date].push(statusResult.status);
        } else {
            finalObj[date] = [statusResult.status];
        }
    })
    const groupArrays = Object.keys(finalObj).map((date) => {
        return {
            date,
            status: finalObj[date]
        };
    });

    let uniqsArr = [];
    for (let i = 0; i < groupArrays.length; i++) {
        var uniqs = groupArrays[i].status.reduce((acc, val) => {
            acc[val] = acc[val] === undefined ? 1 : acc[val] += 1;
            return acc;
        }, {});

        uniqsArr.push(uniqs);

    }
    const groupArrays2 = Object.keys(groupArrays).map((date) => {
        return {
            date: groupArrays[date].date,
            counts: uniqsArr[date]
        };
    });

    let data = [];
    for(let i = 0 ;i <groupArrays2.length; i++) {
        data.push({ name: groupArrays2[i].date, "Available": groupArrays2[i].counts.Available, "Taken": groupArrays2[i].counts.Taken, "Broken": groupArrays2[i].counts.Broken, "Deprecated": groupArrays2[i].counts.Deprecated })
            
    }

    return (
        <div classname="chart-section">
            <AreaChart width={950} height={300} data={data} className="chart-image">
                <CartesianGrid />
                <XAxis dataKey="name"/>
                <YAxis/>
                <Tooltip />
                <Area dataKey="Available" stackId="1"
                    stroke="#55cd97" fill="#55cd97" />
                <Area dataKey="Taken" stackId="1"
                    stroke="#1ca1c1" fill="#1ca1c1" />
                <Area dataKey="Broken" stackId="1" stroke="#ff5c4c" fill="#ff5c4c"></Area>
                <Area dataKey="Deprecated" stackId="1" stroke="#94A1B3" fill="#94A1B3"></Area>
            </AreaChart>
        </div>
    )
}

export default ChartSection