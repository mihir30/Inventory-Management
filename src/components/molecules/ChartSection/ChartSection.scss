.chart-section{
    border-width: 0px;
    margin-left: 0px;
    margin-top: 0px;
    width: 699px;
    height: 300px;
    font-size: 14px;
}

.chart-image{
    width:1000px;
    margin: 35px;
    font-size: 14px;
}


