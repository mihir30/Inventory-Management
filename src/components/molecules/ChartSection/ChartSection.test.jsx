import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import ChartSection from './ChartSection';

describe("Chart Section Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><ChartSection statusResults={[
            {
                "id": 2,
                "status": "Available",
                "serial": "PC-00017",
                "type": "Computer",
                "name": "Dell X2",
                "person": "Barbara Spears",
                "fromDate": "2020-08-08"
            },
            {
                "id": 5,
                "status": "Available",
                "serial": "PC-00018",
                "type": "Computer",
                "name": "Dell X2",
                "person": "Bailey Kerr",
                "fromDate": "2020-09-08"
            }
        ]}></ChartSection></BrowserRouter>, divEl); 
        
    });


    it('should have the right chart section component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});