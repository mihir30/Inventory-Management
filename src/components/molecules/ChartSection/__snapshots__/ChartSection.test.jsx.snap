// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Section Component should have the right chart section component snapshot 1`] = `null`;
